hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruBlockCache.java

```diff
@@ -522,16 +522,9 @@ public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching, boolean repea
       if (victimHandler != null && !repeat) {
         // The handler will increase result's refCnt for RPC, so need no extra retain.
         Cacheable result = victimHandler.getBlock(cacheKey, caching, repeat, updateCacheMetrics);
-
         // Promote this to L1.
         if (result != null) {
           if (caching) {
-            if (result instanceof HFileBlock && ((HFileBlock) result).usesSharedMemory()) {
```

Comment:
As part of the new issue, we have to handle here also.. Because when it is shared memory, we can not keep the block as is in LRU cache.

Reply:
Here, the following cacheBlock in LruBlockCache,   we have a asReferencedHeapBlock method which will deepClone the shared memory into a heap one and cache the heap one.  so there we no need to do extra check & deepClone now, it's safe to remove this.

https://github.com/apache/hbase/pull/257/files/800726459ced6d868abcfd871aa75f0e60c6d275#diff-e300bc9c680e6ccde942561c4c0b5a3dR528